airbyte-cdk/java/airbyte-cdk/typing-deduping/src/main/kotlin/io/airbyte/integrations/base/destination/operation/StorageOperation.kt

```diff
@@ -16,9 +15,38 @@ interface StorageOperation<Data> {
      */
 
     /**
-     * Prepare staging area which cloud be creating any object storage, temp tables or file storage
+     * Prepare staging area which cloud be creating any object storage, temp tables or file storage.
+     * Similar to [createFinalTable], accepts a [suffix] parameter, which should be used in conjunction
+     * with [overwriteStage].
+     *
+     * @param replace If true, then replace existing resources with empty e.g. tables. If false,
+     * then leave existing resources untouched.
      */
-    fun prepareStage(streamId: StreamId, destinationSyncMode: DestinationSyncMode)
+    fun prepareStage(streamId: StreamId, suffix: String, replace: Boolean = false)
```

Comment:
hm. just realized that this is weirdly inconsistent

* prepareStage (and friends) all take a `suffix`
* but `writeToStage` expects the caller to modify the StreamId (i.e. `streamId.copy(rawName = streamId.rawName + suffix)`)

after doing the bigquery implementation, I actually kind of prefer passing an explicit suffix, but not strongly - @airbytehq/destinations wdyt?